lib/OpenQA/Schema/Result/ScheduledProducts.pm

```diff
@@ -563,38 +564,23 @@ sub _generate_jobs {
 
             # variable expansion
             # replace %NAME% with $settings{NAME}
-            my $expanded;
-            do {
-                $expanded = 0;
-                for my $var (keys %settings) {
-                    my $val = $settings{$var};
-                    next unless ($val && $val =~ /(%\w+%)/);
-                    my $replace_var = $1;
-                    $replace_var =~ s/^%(\w+)%$/$1/;
-                    my $replace_val = $settings{$replace_var};
-                    next unless (defined $replace_val);
-                    $replace_val = '' if $replace_var eq $var;    #stop infinite recursion
-                    $val =~ s/%${replace_var}%/$replace_val/g;
-                    $settings{$var} = $val;
-                    $expanded = 1;
-                }
-            } while ($expanded);
+            my $new_settings = OpenQA::Setting->new(%settings)->replace_setting();
```

Comment:
What's the replace meant to do here? Should this just be `OpenQA::Setting->new(%settings)`?

Reply:
The replace will to the actual replacement. Using an object for this although a free function would do is a overkill and a bit confusing.

Reply:
My comment is about semantics actually. I wasn't very clear there. It reads odd to have `new->replace_setting`. But the method rename as per the other conversation in the review would address that.

Reply:
fixed. changed the function name to expand_placeholders()